src/GraphQL/Conversion/ListConverterFactories/CustomListConverterFactory.cs

```diff
@@ -0,0 +1,153 @@
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace GraphQL.Conversion;
+
+/// <summary>
+/// Dynamically constructs a list of the specified type using reflection.
+/// The type must have a public constructor that takes a single parameter compatible
+/// with an array of the element type, such as <see cref="IEnumerable{T}"/>,
+/// or a public parameterless constructor and a public method named "Add" that
+/// takes a single parameter of the element type.
+/// Alternatively, the type may implement <see cref="IList"/> and have a public
+/// parameterless constructor.
+/// </summary>
+internal sealed class CustomListConverterFactory : IListConverterFactory
```

Comment:
This class replaces the previous reflection-based approach available in previous versions of GraphQL.NET.  While this works for Queue and Stack, it does not work for the Immutable classes because (1) there is no public constructor that accepts `IEnumerable<T>`, and (2) there is no public default constructor along with a public method `Add<T>(T item)`.